### DISCUSSION TITLE
Is there are better way to do --embed-resources feature for --to jats?

### DISCUSSION THREAD

**Discussion by castedo** [jgm/pandoc#9798] [Q&A] [answered]
I am embedding base 64 encoded images inside JATS XML by doing the following shell pipe:

```
pandoc --from markdown --embed-resources --to html | pandoc --from html -s --to jats
```

This results is XML like this:
```
  <graphic mimetype="image" mime-subtype="" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgA...
```

Is there better way to achieve this? Perhaps by copying and reusing an existing lua script?

The `--embed-resources` option does not work with `--to jats`.

**Comment by tarleb** [comment, marked as answer]
Here'd be my approach. My quick `base64.encode` hack should probably be replaced with a proper Lua encoder, of which there are quite a few out there.

``` lua
local base64 = {}
base64.encode = function (data)
  return pandoc.pipe('base64', {'--wrap=0'}, data)
end

function Image (img)
  local mime, contents = pandoc.mediabag.fetch(img.src)
  img.src = 'data:' .. mime .. ';base64,' .. base64.encode(contents)
  return img
end
```
